The Hierophant

ChangelingRain opened the pull request:
New boss, "inspired by"(taken from) Hyper Light Drifter.
- [x] Ruin for boss spawning(don't fight in in the ruin, or do I guess? If you fight it in the ruin there's a good chance you die horribly)
- [x] Reward tuning so the staff is useful as a weapon/mining tool.
- [x] Better sounds for the boss's attacks, possibly distinct sounds for different attacks?
  - [x] Note that the above, while done, isn't implemented, because the sounds involved don't make sense for the current sprites.
- [x] Tuning so it's at least somewhat difficult to fight but not overly difficult, so it can be killed for a decent weapon against other bosses.
- [x] ~~Actual sprites for the boss itself(instead of literally the boss's sprite from Hyper Light Drifter)~~
  - [x] ~~Actual sprites for the weapon it drops(instead of a placeholder that looks like the aforementioned boss's staff)~~
  - [x] Rip those goals it's bird time

ChangelingRain committed:
HIEROPHANT

ChangelingRain committed:
whoops

ChangelingRain committed:
that'd be hilarious but unwise

ChangelingRain committed:
BIRD_SCORE

PKPenguin321 commented:
http://hyperlightdrifter.wikia.com/wiki/The_Hierophant

ChangelingRain commented:
yeah it's that fuck but slightly more mean and it doesn't have any vulture mobs with it

ChangelingRain committed:
density

Kromgar commented:
Ergo already killed it

ChangelingRain committed:
BIRD THREAT

ChangelingRain committed:
maybe a threat tm

ChangelingRain committed:
BIRD CONFLICTS
with myself

ChangelingRain committed:
danger boss

ChangelingRain committed:
it's always the paren

ChangelingRain committed:
bird murder

ChangelingRain committed:
recall

ChangelingRain committed:
no message

ChangelingRain committed:
yelling bird

ChangelingRain committed:
time
time
come on, time
time
shades

ChangelingRain committed:
A ruin!

ChangelingRain committed:
I think five blasts is enough

ChangelingRain committed:
birdboss wants you dead

ChangelingRain committed:
giant boss arena

ChangelingRain committed:
hierophant no longer literally unfightable

ChangelingRain committed:
smaller ruin?

ChangelingRain committed:
i'm not sure what changed but okay

ChangelingRain committed:
no longer innate

ExcessiveUseOfCobblestone commented:
Testmerge Feedback:

- Was Enjoyable to fight and was an interesting boss approach. We fought IN the ruins and we did fine [2 people and a minebot].
- Would like some cosmetic heirophant items to cosplay as the boss I just slayed :dress: 
- Staff is more of a liability unless you're going solo, as unlike the other weapons the staff can create weird patterns without you doing it. If there was some option [possibly tied to a heirophant cosmetic like the hat so you aren't spammed with buttons just for holding the staff] that lowered damage against miners and minebots, that would be appreciated.
- The teleport feature is also solo-oriented, despite the aesthetic that makes it look like a group teleport [which is what i'd prefer anyways]. Sorta misleading [unless it's a reference to the teleport animation from HLD, I never played the game].

Other than that it's a great feature and I hope you continue working on it!

ChangelingRain committed:
can toggle friendly fire, teleport is AoE

ChangelingRain commented:
The staff can now toggle friendly fire, and the teleport will teleport all mobs near you.
Those were good ideas, so I've implemented them.

Clothes not included, because the boss's sprite is absolutely not final and I'd rather not have to resprite the clothes to fit whatever the boss becomes.

Repukan commented:
So many flashing lights, either my internet is shit or the amount of light worms and patterns caused lag.

All in all, pretty fun to play against. Tried soloing it with a proto and got my ass kicked. 10/10 nerf the epilepsy

ChangelingRain commented:
Lag may or may not be shitcode, I'm not sure how optimized the code this is using is.

ChangelingRain committed:
comments!

ChangelingRain committed:
just fucking spawn the arena wew

ChangelingRain committed:
heal on resetting

ChangelingRain commented:
Hierophant will now heal for 25% of its missing health if it times out and resets. Letting it reset is bad and should not be abused, mmkay?

ChangelingRain committed:
not as common

ChangelingRain committed:
I cannot be trusted

ChangelingRain committed:
eeeh

ChangelingRain committed:
!

ChangelingRain committed:
keep it off the station

Kromgar commented:
>not letting boss mobs go onto station

Why do you hate fun?

PKPenguin321 commented:
yeah thats lame

ChangelingRain commented:
If you can get it on the station it'll still be a huge boss but the crew can just try and ignore it and it'll leave if they manage that for long enough. Keywork 'if'

ChangelingRain committed:
as common as sin

ChangelingRain committed:
stop cheesing the boss you asshole

ChangelingRain commented:
Fixed TK boss cheese(which actually worked on literally every boss provided you kept it right at the edge of sight range)

ChangelingRain committed:
this looks better

ChangelingRain committed:
that ruin didn't even exist what the fuck

ChangelingRain committed:
wew

ChangelingRain committed:
rip

ChangelingRain committed:
it doesn't need full healing if it heals for half when it resets

ChangelingRain committed:
good for fighting mobs

ChangelingRain committed:
rune where you are

ChangelingRain committed:
mechanics; armor doesn't fucking matter rip you

ChangelingRain committed:
armor matters but not much so

ChangelingRain committed:
)

ChangelingRain committed:
meaner than hell and more likely to kill you

ChangelingRain committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into broadcastwhoyouareonchannelsnobodyusesinhopessomethinghearsitbeahierophanttonothing

ChangelingRain committed:
remove those beasts from my beautiful square house

ChangelingRain committed:
fix

optimumtact commented:
stale?

ChangelingRain commented:
NOT STALE

ChangelingRain committed:
NOT STALE

ChangelingRain committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into broadcastwhoyouareonchannelsnobodyusesinhopessomethinghearsitbeahierophanttonothing

Conflicts:
	icons/mob/actions.dmi

Kromgar commented:
I love the Hierophants Staff of Memes

ChangelingRain commented:
There is an achievement for killing it but the scores/etc from the testing will not carry over to when it's merged.

ChangelingRain committed:
you should be able to tell what it's doing, even if you can't actually react to it
being totally unable to tell what the fuck just killed you is bad design, you should know how you died even if you couldn't do shit
rip 3x3 chaser

ChangelingRain committed:
start higher scale to the same

ChangelingRain committed:
I KEEP MERGING THINGS THAT CONFLICT WITH THIS

ChangelingRain committed:
the ruin that doesn't exist and costs 10 points to place

ChangelingRain committed:
now what's all this commented out stuff hmmmmmmmmmmm
what could it possibly be

optimumtact commented:
Is this ready yet bloody hell

ChangelingRain commented:
It's waiting on sprites(possibly from kor) and is ready besides those and I'll get super mad if you close it.

optimumtact commented:
Reopen when you have the sprites.

ChangelingRain committed:
accurate

Kromgar commented:
@ChangelingRain Jesus fucking christ. What did you do to it? I took it down a couple of days ago was pretty hard.

Faced it as a changeling with survival pens an upgraded ka and a plasma blaster. Suddenly near its end phase where it starts teleporting around.


BOOOM BOOM BOOOM nonstop its summoning massive aoe circles ontop of me. Considering the arena and the size and how rapidly it was putting things out i died with survival pens and fleshmend on my side.

ChangelingRain commented:
The arena is basically hardmode for it, given how large the aoe blasts get it's sometimes not possible to dodge if you're cornered; this is totally intended if you want to fight it and die less I suggest luring it out of the arena.

ChangelingRain committed:
tracking

ChangelingRain committed:
boring unused lore

ChangelingRain commented:
It turns out this pr needs to be open to push updates to it for testmerges, as the version on the server should have `Stolen from Hyper Light Drifter. Commit number 58.` as the hierophant's description(the server had this testmerged after the commit that would have done that) and it doesn't.
RIP orange's tracker, i guess.

ChangelingRain committed:
it needs to be open, oranges. open.

ChangelingRain committed:
I did that specifically to note which commit it was on ree

TechnoAlchemisto commented:
The sprites from the game don't look that out of place to be quite honest.

ChangelingRain commented:
Yeah but it's literally ripped from the game and I don't want to literally just take the sprites from the game.

Kromgar commented:
Copyright theft is not a good thing

ChangelingRain committed:
I do everything I can to conflict with myself

ChangelingRain committed:
At least I know how to handle conflicts so this isn't a pain in the butt

ChangelingRain committed:
oh there's a proc for that. duh.

ChangelingRain committed:
logging is good

ChangelingRain commented:
Oh hell fuck it I'll get sprites eventually and I can make it what it's supposed to be then(it's supposed to be an alien death robot with some sort of weapon but this huge fucking bird will have to do)

ChangelingRain committed:
caw

TechnoAlchemisto commented:
:tada:

Kromgar commented:
>Makes an arena too small to fight the hierophant in

Whats the point of an arena if its too small?

Why does it spawn a 3x3 directly on top of the player that has no travel time? 


Its literally unkillable at this point you realize that right?

ChangelingRain committed:
horrifying

Kromgar commented:
IF you could just make the arena entirely empty save for 4 pillars the arena might actually be viable to fight the hierophant in

ChangelingRain committed:
the marginal kindness of a giant bird boss

ChangelingRain commented:
The thing about the arena is, short of it being gigantic and empty(super boring), any arena a player can make will absolutely be better than the possibility of ever being cornered in this arena.
If you remove the possibility of being cornered, then the arena isn't really an arena, it's just a big open space.
Aaaand I don't really like forcing people to fight it in the arena because A: absolutely not the feel of the game B: >totally removing all possibility of getting it to the station for high tier murder

Or in other words I'm cornered into a 'challenge mode' arena by the nature of the game and my own preferences.

ChangelingRain committed:
ruin tweak

ChangelingRain committed:
idiot bird

ChangelingRain committed:
adjust

optimumtact commented on code/datums/action.dm:
```diff
@@ -189,6 +189,40 @@
+		UpdateButtonIcon()
+
+/datum/action/item_action/toggle_unfriendly_fire/UpdateButtonIcon()
+	if(istype(target, /obj/item/weapon/hierophant_staff))
```
what the fuck is this, shouldn't this be handled by the staff?

ChangelingRain replied:
Having the staff handle it requires a for loop or using locate so this is ~technically better~ even though it's weird
